File: contracts/rust/fairauction/test/consts.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package test

import (
	"github.com/iotaledger/wasp/packages/coretypes"
)

const ScName = "fairauction"
const ScHname = coretypes.Hname(0x1b5c43b1)

const ParamColor = "color"
const ParamDescription = "description"
const ParamDuration = "duration"
const ParamMinimumBid = "minimumBid"
const ParamOwnerMargin = "ownerMargin"

const VarAuctions = "auctions"
const VarBidderList = "bidderList"
const VarBidders = "bidders"
const VarColor = "color"
const VarCreator = "creator"
const VarDeposit = "deposit"
const VarDescription = "description"
const VarDuration = "duration"
const VarHighestBid = "highestBid"
const VarHighestBidder = "highestBidder"
const VarInfo = "info"
const VarMinimumBid = "minimumBid"
const VarNumTokens = "numTokens"
const VarOwnerMargin = "ownerMargin"
const VarWhenStarted = "whenStarted"

const FuncFinalizeAuction = "finalizeAuction"
const FuncPlaceBid = "placeBid"
const FuncSetOwnerMargin = "setOwnerMargin"
const FuncStartAuction = "startAuction"
const ViewGetInfo = "getInfo"

const HFuncFinalizeAuction = coretypes.Hname(0x8d534ddc)
const HFuncPlaceBid = coretypes.Hname(0x9bd72fa9)
const HFuncSetOwnerMargin = coretypes.Hname(0x1774461a)
const HFuncStartAuction = coretypes.Hname(0xd5b7bacb)
const HViewGetInfo = coretypes.Hname(0xcfedba5f)

File: contracts/rust/fairauction/src/consts.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

#![allow(dead_code)]

use wasmlib::*;

pub const SC_NAME: &str = "fairauction";
pub const HSC_NAME: ScHname = ScHname(0x1b5c43b1);

pub const PARAM_COLOR: &str = "color";
pub const PARAM_DESCRIPTION: &str = "description";
pub const PARAM_DURATION: &str = "duration";
pub const PARAM_MINIMUM_BID: &str = "minimumBid";
pub const PARAM_OWNER_MARGIN: &str = "ownerMargin";

pub const VAR_AUCTIONS: &str = "auctions";
pub const VAR_BIDDER_LIST: &str = "bidderList";
pub const VAR_BIDDERS: &str = "bidders";
pub const VAR_COLOR: &str = "color";
pub const VAR_CREATOR: &str = "creator";
pub const VAR_DEPOSIT: &str = "deposit";
pub const VAR_DESCRIPTION: &str = "description";
pub const VAR_DURATION: &str = "duration";
pub const VAR_HIGHEST_BID: &str = "highestBid";
pub const VAR_HIGHEST_BIDDER: &str = "highestBidder";
pub const VAR_INFO: &str = "info";
pub const VAR_MINIMUM_BID: &str = "minimumBid";
pub const VAR_NUM_TOKENS: &str = "numTokens";
pub const VAR_OWNER_MARGIN: &str = "ownerMargin";
pub const VAR_WHEN_STARTED: &str = "whenStarted";

pub const FUNC_FINALIZE_AUCTION: &str = "finalizeAuction";
pub const FUNC_PLACE_BID: &str = "placeBid";
pub const FUNC_SET_OWNER_MARGIN: &str = "setOwnerMargin";
pub const FUNC_START_AUCTION: &str = "startAuction";
pub const VIEW_GET_INFO: &str = "getInfo";

pub const HFUNC_FINALIZE_AUCTION: ScHname = ScHname(0x8d534ddc);
pub const HFUNC_PLACE_BID: ScHname = ScHname(0x9bd72fa9);
pub const HFUNC_SET_OWNER_MARGIN: ScHname = ScHname(0x1774461a);
pub const HFUNC_START_AUCTION: ScHname = ScHname(0xd5b7bacb);
pub const HVIEW_GET_INFO: ScHname = ScHname(0xcfedba5f);
